### ISSUE TITLE
Checksum issues mainly with SMB mount point + Google Drive

### ISSUE THREAD

**Issue by mmattel** [owncloud/core#40809]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

The owncloud instance running 10.11 in combination of the desktop app 3.2 and all version before never had syncing issues, a recent upgrade to desktop 4.0 did not changed the following behaviour.

**Problem:**

Having some SMB and Google Drive mount points and post upgrading to server 10.12.0 respectively 10.12.1, **syncing data fails in the following conditions**, but ONLY when using SMB, works flawless with GD.

Files can be located in any folder/subfolder representing the synced SMB mountpoint:

In these conditions the issue appears:

* Adding a new file via the local filesystem.
* Editing an existing file like with Excel.
* Overwriting an existing file via the **browser**.
Note that the file gets fully accepted by the backend and can be downloaded manually without any issue, but fails syncing.

When the file is NOT present in the local sync folder, you can upload it via browser and it gets synced. It is then available locally as usual

This shows up like a sync issue from the client to the server but not vice versa.

This is the SMB mount point:
![image](https://github.com/owncloud/client/assets/3321281/c963a145-d2eb-4766-9750-5c794619b302)

### Expected behavior

Files should sync to SMB mount points as before

### Steps to reproduce the issue

See the bug description above

### Logs

Right click on files that failed syncing:

Locally present but should get overwritten by the server (file update via browser)
```
Issues         ,File                                                            ,Folder                        ,Size  ,Account             ,Time                ,Status              ,
The downloaded file does not match the checksum, it will be resumed. '374e32ad842ed171ac51af672724887cea3a0c2f' != '98d3eb8c44e936c7d7110c09b4464a91e2c7cfae' (skipped due to earlier error, trying again in 21 second(s)),martin@filer/Dokumente/Firma/Buchhaltung/2023/2023 Abgaben Bewegungen.xlsx,ownCloud\diemattels.at        ,      ,mmattel@owncloud.diemattels.at,2023-05-18T10:06:07.845,Blacklisted         ,
```
Newly added local file
```
Issues         ,File                                                            ,Folder                        ,Size  ,Account             ,Time                ,Status              ,
Server replied "400 Bad Request" to "PUT https://owncloud.diemattels.at/remote.php/dav/files/mmattel/martin@filer/desktop_client_test.txt" (The computed checksum does not match the one received from the client.),martin@filer/desktop_client_test.txt                            ,ownCloud\diemattels.at        ,0 B   ,mmattel@owncloud.diemattels.at,2023-05-18T10:13:36.000,Error               ,
```

Log for newly added local file
```
23-05-18 10:24:51:311 [ info sync.propagator ]:	Starting SyncInstruction(CSYNC_INSTRUCTION_IGNORE) propagation of "martin@filer/desktop_client_test.txt" by OCC::PropagateIgnoreJob(0x2ea7a0d3fb0)
23-05-18 10:24:51:312 [ warning sync.propagator ]:	Could not complete propagation of "martin@filer/desktop_client_test.txt" by OCC::PropagateIgnoreJob(0x2ea7a0d3fb0) with status OCC::SyncFileItem::BlacklistedError and error: "Server replied \"400 Bad Request\" to \"PUT https://owncloud.diemattels.at/remote.php/dav/files/mmattel/martin@filer/desktop_client_test.txt\" (The computed checksum does not match the one received from the client.) (skipped due to earlier error, trying again in 10 minute(s))"
23-05-18 10:24:51:312 [ debug sync.statustracker ]	[ OCC::SyncFileStatusTracker::slotItemCompleted ]:	Item completed "martin@filer/desktop_client_test.txt" OCC::SyncFileItem::BlacklistedError SyncInstruction(CSYNC_INSTRUCTION_IGNORE)
23-05-18 10:24:51:312 [ info gui.socketapi ]:	Sending SocketAPI message --> "STATUS:IGNORE:C:\\Users\\marti\\ownCloud\\diemattels.at\\martin@filer" to QLocalSocket(0x2ea74a11a80)
23-05-18 10:24:51:312 [ info gui.socketapi ]:	Sending SocketAPI message --> "STATUS:ERROR:C:\\Users\\marti\\ownCloud\\diemattels.at\\martin@filer\\desktop_client_test.txt" to QLocalSocket(0x2ea74a11a80)
23-05-18 10:24:51:312 [ debug sync.localdiscoverytracker ]	[ OCC::LocalDiscoveryTracker::slotItemCompleted ]:	inserted error item "martin@filer/desktop_client_test.txt"
```

### Client version number

**Client:**
* Windows x64
* 4.0.0.10896
* QT 5.15.8
* NO virtual files

### Installation path (Windows only)

C:\Program Files\ownCloud

### Server information

**Server:**
* Ubuntu 20.04
* mariadb
* php7.4
* Version 10.12.1

### Additional context

The interesting thing is, that it was challenging to identify that this is a checksum issue because the client tells that the file is blacklisted. As the server has a blacklisting configuration possibility too, this was confusing. Note that there is NO server blacklisting configured!

**Comment by michaelstingl** [comment]
@mmattel @jnweiger @IljaN issue transferred to server repo. IMHO nothing to fix in the desktop app.

**Comment by Helios07** [comment]
I can also see that issue on an instance with owncloud 10.12 and external SMB mounts. Please ping me, if I can deliver any logs, etc.

**Comment by jnweiger** [comment]
I can reproduce a similar issue with an SFTP mount on 10.12.2 -- my SMB mounts work fine.

**Comment by Helios07** [comment]
Interestingly, this happens only for a single or at least some users on that server and I'm not sure, if this was the case at the beginning. Nevertheless, I upgraded from 10.12.0 to 12.12.2 and will observe, if it still is the case.

**Comment by jnweiger** [comment]
I've opened a new clean issue with a bisect for my case: https://github.com/owncloud/core/issues/40831

**Comment by jnweiger** [comment]
@mmattel @Helios07 chances are, that #40831 already does the trick here too ...

**Comment by mmattel** [comment]
Adding the changes of the referenced PR manually fixed the problem and syncing is possible again.

**Comment by jnweiger** [comment]
Thanks for confirming.
@Helios07 does that work for you too?

**Comment by jnweiger** [comment]
Thanks for double checking!
Closing here, all is in master.
Patch URL for reference: https://patch-diff.githubusercontent.com/raw/owncloud/core/pull/40832.patch

**Comment by jnweiger** [comment]
Confirmed fixed in 10.13.0-rc.1
